src/charm.py

```diff
@@ -176,6 +186,29 @@ def _on_cluster_relation_changed(self, event: EventBase) -> None:
         if self.state.cluster.switching_encryption and len(self.state.servers) == 1:
             event.defer()
 
+    def _on_secret_changed(self, event: SecretChangedEvent):
+        """Reconfigure services on a secret changed event."""
+        if not event.secret.label:
+            return
+
+        if not self.state.cluster.relation:
+            return
+
+        if event.secret.label == self.state.cluster.data_interface._generate_secret_label(
```

Comment:
**question:** What does this actually check? Ignorantly, to me it just seems like it's checking if a string == string.

Reply:
I ask that, because generally we're trying to funnel every event in to `cluster-relation-changed`. To me, there isn't a distinction between a 'secret-changed' and a 'relation-changed', conceptually I mean. 

Am I fair in guessing that we can still observe secret-changed, and then call the relation-changed handler as normal?

Reply:
We check _which_ secret changed.

Precisely since we don't actually care on a service level, if the `unit` TLS secrets changed. While we DO care on a service level whether the `sync-password` changed -- we need to cascade that down to the Zookeeper application.

So, while primarily you are right -- it's "just a `relation-changed` thing", we still need to have extra, secrets-specific checks on this occasion.

I'm wondering that it may rather worth to group functionalities that may happen both on secret changes AND on databag changes into an `update_service` function, which is called both in `relation-changed` and in `secret-changed`.

Does the response make sense...?

Reply:
OK, can we:
1. Comment the `if not event.secret.label` to explain what it's for.
2. Comment the `if event.secret.label == self.state.cluster.data_interface._generate_secret_label()` line to explain why it's only app secrets we care about
3. Comment the docstring for the `_on_secret_changed` explaining why we handle this specifically (i.e the slow request time for secret content)
4. Replace the `if:` below to just call `self._on_cluster_relation_changed(event)`